pkg/odo/genericclioptions/runnable.go

```diff
@@ -22,12 +23,42 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
 	// fixes / checks all related machine readable output functions
 	CheckMachineReadableOutputCommand(cmd)
 
+	// LogErrorAndExit is used so that we get -o (jsonoutput) for cmds which have json output implemented
+	util.LogErrorAndExit(checkConflictingFlags(cmd), "")
 	// Run completion, validation and run.
 	util.LogErrorAndExit(o.Complete(cmd.Name(), cmd, args), "")
 	util.LogErrorAndExit(o.Validate(), "")
 	util.LogErrorAndExit(o.Run(), "")
 }
 
+func checkConflictingFlags(cmd *cobra.Command) error {
+
+	// we allow providing --context with --app and --project in case of `odo create` or `odo component create`
+	if cmd.Name() == "create" {
+		if cmd.HasParent() {
+			if cmd.Parent().Name() == "odo" || cmd.Parent().Name() == "component" {
+				return nil
+			}
+		}
+	}
+	app := stringFlagLookup(cmd, "app")
+	project := stringFlagLookup(cmd, "project")
+	context := stringFlagLookup(cmd, "context")
+	component := stringFlagLookup(cmd, "component")
+	if (context != "") && (app != "" || project != "" || component != "") {
```

Comment:
read if `context flag is provided and with that either of app, project or component are also provided` then return error.